feat: pass extauthz metadata to Bedrock requestMetadata

apexlnc opened the pull request:
Flattens nested extauthz JSON into Bedrock's requestMetadata field for observability and log filtering. AWS validates constraints server-side.

howardjohn commented on crates/agentgateway/src/llm/bedrock.rs:
```diff
@@ -120,6 +121,32 @@ pub fn process_response(
 	}
 }
 
+/// Flatten nested JSON into simple key-value pairs for Bedrock requestMetadata.
```
Would it make sense to instead allow a specific key to be set. something like `agentgateway.bedrock.metadata` or something?

howardjohn replied:
I don't think passing all of the metadata blindly is appropriate; the server may be putting anything in there, from secrets to other data.

apexlnc replied:
how do we feel about using CEL transformation here? seems a bit "extra" , but curious on your thoughts cc @howardjohn 

```
  transformation:
    request:
      set:
        - name: x-bedrock-metadata
          value: 'to_json({"user_id": extauthz["extauthz.pat"].user_id})'
 ```

howardjohn replied:
Hmm. In general I like the idea of a CEL expression being used since its the most flexible. The problem with the above is  transformation is, by definition, setting a header. But I think bedrock metadata is in the json body(?)

I guess this would just mean we actually read the header and map it into the body which seems reasonable. It would allow a user to set it directly which may or may not be a good thing

apexlnc replied:
> by definition, setting a header. But I think bedrock metadata is in the json body(?) 

yeah, that's why I said this was "extra" and was me trying real hard to use CEL + give users knobs to control what fields they want to extract. 

my ORIGINAL plan was to expose `requestMetadata` as either a `policy` or provider config (like `guardrails`), but decided not to

apexlnc replied:
if you're fine with this @howardjohn , the flexibility with this approach is nice.

apexlnc committed:
feat: pass ExtAuthz metadata to Bedrock requestMetadata via CEL

Add x-bedrock-metadata header extraction to pass ExtAuthz metadata to Bedrock's
requestMetadata field. Supports CEL transformations with to_json() to convert
CEL maps to JSON strings for header values.

Example:
  transformation:
    request:
      set:
        - name: x-bedrock-metadata
          value: 'to_json({"user_id": extauthz["extauthz.pat"].user_id})'

howardjohn reviewed (approved):
thanks!
